handlers/accesscontrol/access-control.go

```diff
@@ -79,12 +100,31 @@ func periodicCleanUpHitRecordCache() {
 	}()
 }
 
-func NewAccessControlHandlersCollection(cli config.Cli) *AccessControlHandlersCollection {
+func periodicRefreshIntervalCache(mapic mistapiconnector.IMac, accessControlCache map[string]map[string]*PlaybackAccessControlEntry) {
+	go func() {
+		for {
+			time.Sleep(time.Duration(5) * time.Second)
+			for key := range refreshIntervalCache.data {
+				if time.Since(refreshIntervalCache.data[key].LastRefresh) > time.Duration(refreshIntervalCache.data[key].RefreshInterval)*time.Second {
+					refreshIntervalCache.data[key].LastRefresh = time.Now()
+					mapic.InvalidateAllSessions(key)
+					playbackIdCache := accessControlCache[key]
+					for cacheKey := range playbackIdCache {
+						delete(playbackIdCache, cacheKey)
+					}
+				}
+			}
+		}
+	}()
+}
 
-	periodicCleanUpHitRecordCache()
+func NewAccessControlHandlersCollection(cli config.Cli, mapic mistapiconnector.IMac) *AccessControlHandlersCollection {
+	accessControlCache := make(map[string]map[string]*PlaybackAccessControlEntry)
+	periodicCleanUpRecordCache()
+	periodicRefreshIntervalCache(mapic, accessControlCache)
```

Comment:
maybe we should just make this function part of AccessControlHandlersCollection then we can just access the cache field in the normal way.
e.g `func (ac *AccessControlHandlersCollection) periodicRefreshIntervalCache(...`
then we can access `ac.cache`